lib/services/access_request.go

```diff
@@ -60,6 +60,44 @@ func ValidateAccessRequest(ar types.AccessRequest) error {
 	return nil
 }
 
+// ClusterGetter provides access to the local cluster
+type ClusterGetter interface {
+	// GetClusterName returns the local cluster name
+	GetClusterName(opts ...MarshalOption) (types.ClusterName, error)
+	// GetRemoteCluster returns a remote cluster by name
+	GetRemoteCluster(clusterName string) (types.RemoteCluster, error)
+}
+
+// ValidateAccessRequestClusterNames checks that the clusters in the access request exist
+func ValidateAccessRequestClusterNames(cg ClusterGetter, ar types.AccessRequest) error {
+	localClusterName, err := cg.GetClusterName()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	var errs []error
+	for _, resourceID := range ar.GetRequestedResourceIDs() {
+		if resourceID.ClusterName == "" {
+			continue
+		}
+		if resourceID.ClusterName == localClusterName.GetClusterName() {
+			continue
+		}
+		_, err := cg.GetRemoteCluster(resourceID.ClusterName)
+		if err == nil {
+			continue
+		}
+		if trace.IsNotFound(err) {
+			errs = append(errs, trace.NotFound("access request contains invalid cluster name %q", resourceID.ClusterName))
```

Comment:
Don't mean to be too pedantic but instead of returning a list of "access request contains invalid cluster name" errors for each invalid name, can we return a single error with a list of invalid cluster names? Basically, collect invalid cluster names instead of collecting multiple and returning aggregate.